RestartManager.js

```diff
@@ -3,41 +3,54 @@ const NativeCodePush = require("react-native").NativeModules.CodePush;
 
 const RestartManager = (() => {
     let _allowed = true;
-    let _restartPending = false;
+    let _restartInProgress = false;
+    let _restartQueue = [];
 
     function allow() {
         log("Re-allowing restarts");
         _allowed = true;
-        
-        if (_restartPending) {
+
+        if (_restartQueue.length) {
             log("Executing pending restart");
-            restartApp(true);
+            restartApp(_restartQueue.shift(1));
         }
     }
-    
+
     function clearPendingRestart() {
-        _restartPending = false;
+        _restartQueue = [];
     }
-    
+
     function disallow() {
         log("Disallowing restarts");
         _allowed = false;
     }
 
-    function restartApp(onlyIfUpdateIsPending = false) {
-        if (_allowed) {
-            NativeCodePush.restartApp(onlyIfUpdateIsPending);
-            log("Restarting app");
-        } else {
+    async function restartApp(onlyIfUpdateIsPending = false) {
+        if (_restartInProgress) {
+            log("Restart request queued until the current restart is completed");
+            _restartQueue.push(onlyIfUpdateIsPending);
+        } else if (!_allowed) {
             log("Restart request queued until restarts are re-allowed");
-            _restartPending = true;
-            return true;
+            _restartQueue.push(onlyIfUpdateIsPending);
+        } else {
+            _restartInProgress = true;
+            log("Restarting app");
```

Comment:
Does the interior of that if block actually get executed if the app is restarted?